GoogleDataTransport/GDTCORLibrary/GDTCORStorage.m

```diff
@@ -132,11 +131,9 @@ - (void)storeEvent:(GDTCOREvent *)event {
       }
     }
 
-    // If running in the background, save state to disk and end the associated background task.
-    if (bgID != GDTCORBackgroundIdentifierInvalid) {
-      [[GDTCORApplication sharedApplication] endBackgroundTask:bgID];
-      bgID = GDTCORBackgroundIdentifierInvalid;
-    }
+    // Cancel or end the associated background task if it's still valid.
+    [[GDTCORApplication sharedApplication] endBackgroundTask:bgID];
```

Comment:
I did remove the check here and a few other places since `endBackgroundTask` validates that the identifier is valid, but I can add it back if requested.